clients/privacy-center/cypress/e2e/consent-banner.cy.ts

```diff
@@ -0,0 +1,233 @@
+            .contains("This test website is overriding the banner description label.");
+          cy.get("div#fides-consent-banner-buttons.fides-consent-banner-buttons").within(() => {
+            cy.get("button#fides-consent-banner-button-tertiary.fides-consent-banner-button.fides-consent-banner-button-tertiary")
+              .contains("Customize Test");
+            cy.get("button#fides-consent-banner-button-secondary.fides-consent-banner-button.fides-consent-banner-button-secondary")
+              .contains("Reject Test");
+            cy.get("button#fides-consent-banner-button-primary.fides-consent-banner-button.fides-consent-banner-button-primary")
+              .contains("Accept Test");
+
+            // Order matters - it should always be tertiary, then secondary, then primary!
+            cy.get("button").within(buttons => {
+              const buttonIds = buttons.map((_, elem) => elem.id).get();
+              cy.wrap(buttonIds).should("have.ordered.members", [
+                "fides-consent-banner-button-tertiary",
+                "fides-consent-banner-button-secondary",
+                "fides-consent-banner-button-primary",
+              ]);
+            });
+          })
+        });
+      });
+
+      it("should allow accepting all", () => {
+        cy.contains("button", "Accept Test").should("be.visible").click();
+        // TODO: consent.cy.ts uses cy.waitUntil to wait longer for the cookie to save in CI - will this be needed?
+        cy.getCookie(CONSENT_COOKIE_NAME).should("exist").then((cookie) => {
+          const cookieKeyConsent = JSON.parse(
+            decodeURIComponent(cookie!.value)
+          );
+          expect(cookieKeyConsent).property("data_sales").is.eql(true);
+          expect(cookieKeyConsent).property("tracking").is.eql(true);
+        })
+        cy.contains("button", "Accept Test").should("not.be.visible");
+      });
+
+      it("should support rejecting all consent options", () => {
+        cy.contains("button", "Reject Test").should("be.visible").click();
+        // TODO: consent.cy.ts uses cy.waitUntil to wait longer for the cookie to save in CI - will this be needed?
+        cy.getCookie(CONSENT_COOKIE_NAME).should("exist").then((cookie) => {
+          const cookieKeyConsent = JSON.parse(
+            decodeURIComponent(cookie!.value)
+          );
+          expect(cookieKeyConsent).property("data_sales").is.eql(false);
+          expect(cookieKeyConsent).property("tracking").is.eql(false);
+        })
+      });
+
+      it("should navigate to Privacy Center to manage consent options", () => {
+        cy.contains("button", "Customize Test").should("be.visible").click();
+        cy.url().should("equal", "http://localhost:3000/");
+      });
+
+      it.skip("should save the consent request to the Fides API", () => {
+        // TODO: add tests for saving to API (ie PATCH /api/v1/consent-request/{id}/preferences...)
+        expect(false).is.eql(true);
+      });
```

Comment:
Supporting saving consent requests to the API from the banner itself will require larger changes to `fides-consent.ts` because all the interactions with the Fides API are currently in the Privacy Center code (and gated behind potential identity verification steps, etc.). So we'll need to think about how to design this...